lib/spack/spack/build_systems/python.py

```diff
@@ -403,3 +407,66 @@ def import_module_test(self):
 
     # Check that self.prefix is there after installation
     run_after('install')(PackageBase.sanity_check_prefix)
+
+    def view_file_conflicts(self, view, merge_map):
+        """Report all file conflicts, excepting special cases for python.
+           Specifically, this does not report errors for duplicate
+           __init__.py files for packages in the same namespace.
+        """
+        conflicts = list(dst for src, dst in merge_map.items()
+                         if os.path.exists(dst))
+
+        if conflicts and self.py_namespace:
+            ext_map = view.extensions_layout.extension_map(self.extendee_spec)
+            namespaces = set(
+                x.package.py_namespace for x in ext_map.values())
+            namespace_re = (
+                r'site-packages/{0}/__init__.py'.format(self.py_namespace))
+            find_namespace = match_predicate(namespace_re)
+            if self.py_namespace in namespaces:
+                conflicts = list(
+                    x for x in conflicts if not find_namespace(x))
+
+        return conflicts
+
+    def add_files_to_view(self, view, merge_map):
```

Comment:
I like the overridability of this but it could be more declarative.  It seems like anyone who overrides `add_files_to_view` has to duplicate a lot of traversal logic, but they really just want to say "do some special things  (like copy, filter, etc.) for a few special cases, and do the regular thing in other cases".  I think it's worth thinking about ways to make this clearer in a later PR so that packages don't have to write as much code.